.github/workflows/ci.yml

```diff
@@ -143,10 +145,15 @@ jobs:
         run: sed 's/enforceTimeLimit="true"/enforceTimeLimit="false"/g' phpunit.xml.dist > phpunit.xml
 
       - name: Run tests
+        if: matrix.calculate-code-coverage != 'yes'
         env:
           PHP_CS_FIXER_IGNORE_ENV: ${{ matrix.PHP_CS_FIXER_IGNORE_ENV }}
           FAST_LINT_TEST_CASES: ${{ matrix.FAST_LINT_TEST_CASES }}
-        run: vendor/bin/phpunit ${{ matrix.phpunit-flags || '--exclude-group auto-review' }}
+        run: vendor/bin/paraunit run ${{ matrix.phpunit-flags || '--exclude-group auto-review' }}
+
+      - name: Collect code coverage
+        if: matrix.calculate-code-coverage == 'yes'
+        run: vendor/bin/paraunit coverage --testsuite coverage --exclude-group covers-nothing --clover build/logs/clover.xml
```

Comment:
did you tried to compare results of coverage generated by PHPUnit and by paraunit? what's the difference?

Reply:
No, I did not verify it, but I'll do it 👍.

Reply:
In terms of coverage it looks like everything works as it should:

- [Coverage:  93.15% (23631/25368)](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/5438902047/jobs/9890414466#step:12:522) for #5892
- [Coverage:  94.47% (23942/25344)](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/actions/runs/5438857989/jobs/9890348970?pr=6883#step:13:519) for this MR

Reply:
i have the feeling that number should not be different.

Reply:
The number is different because the code is having global state, e.g. https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v3.21.1/src/Tokenizer/CT.php#L99-L104 - running it with PHPUnit make that `if` covered (or not covered) all the time with the same result, ParaUnit make it non-deterministic.

If you run PHPUnit like `./vendor/bin/phpunit --order-by=random` you will get slightly different coverage for each run.